src/main/java/me/drepic/proton/RedisManager.java

```diff
@@ -0,0 +1,156 @@
+package me.drepic.proton;
+
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.pubsub.RedisPubSubListener;
+import io.lettuce.core.pubsub.StatefulRedisPubSubConnection;
+import io.lettuce.core.pubsub.api.sync.RedisPubSubCommands;
+import me.drepic.proton.message.MessageAttributes;
+import me.drepic.proton.message.MessageContext;
+import me.drepic.proton.redis.RedisChannel;
+import me.drepic.proton.redis.RedisDataWrapper;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+public class RedisManager extends ProtonManager {
+
+    String host;
+    String password;
+    int port;
+
+    RedisClient client;
+    RedisPubSubCommands<String, String> subCommands;
+    RedisPubSubCommands<String, String> pubCommands;
+    StatefulRedisPubSubConnection<String, String> subConnection;
+    StatefulRedisPubSubConnection<String, String> pubConnection;
+
+    protected RedisManager(String name, String[] groups, String host, int port, String password) {
+        super(name, groups);
+        this.host = host;
+        this.port = port;
+        this.password = password;
+        connect();
+    }
+
+    protected RedisManager(String name, String[] groups, String host, int port) {
+        this(name, groups, host, port, "");
+    }
+
+    @Override
+    protected void connect() {
+        if(this.password.isEmpty()){
+            this.client = RedisClient.create(RedisURI.Builder.redis(this.host, this.port).build());
+        }else{
+            this.client = RedisClient.create(RedisURI.Builder.redis(this.host, this.port).withPassword(this.password.toCharArray()).build());
+        }
+
+        subConnection = client.connectPubSub();
+        pubConnection = client.connectPubSub();
+        subConnection.addListener(new RedisPubSubListener<String, String>() {
+            @Override
+            public void message(String channel, String data) {
+                deliveryCallback(channel, data);
+            }
+
+            @Override
+            public void message(String s, String k1, String s2) {}
+
+            @Override
+            public void subscribed(String s, long l) {}
+
+            @Override
+            public void psubscribed(String s, long l) {}
+
+            @Override
+            public void unsubscribed(String s, long l) { }
+
+            @Override
+            public void punsubscribed(String s, long l) {}
+
+        });
+
+        subCommands = subConnection.sync();
+        pubCommands = pubConnection.sync();
+
+    }
+
+    protected void deliveryCallback(String channelString, String json){
+        RedisChannel channel = gson.fromJson(channelString, RedisChannel.class);
```

Comment:
Why is this stored as json instead of just the three strings separated by periods?